### DISCUSSION TITLE
Asset store sync with git

### DISCUSSION THREAD

**Discussion by noisymime** [requarks/wiki#3354] [Help / Questions]
**Question**
I did an initial sync with a github repo that contains both pages and images (all images being located in a single `img` folder in the repo). This appears to have brought in the images from the repo, which is terrific, but am I correct in thinking this is a one time event rather than something that is continuously synced like the pages? If I update the image on github, it doesn't appear that it gets updated in the asset store. 

Is this the expected behavior? 

**Host Info (please complete the following information):**
OS: Ubuntu 19.04, Docker
Wiki.js version: 2.0.12
Database engine: mysql

**Comment by NGPixel** [comment]
Both pages and assets are updated during sync. After you updated the image on GitHub, have you waited for the next sync to occur first? Changes are only processed every 5 mins (by default).

You can also try to clear the cache in the admin > utilities.

**Comment by noisymime** [comment]
Thanks for the reply, I think I've figured out what was going on here. When I did the initial github sync, these images were all in the root directory of the repo. Subsequently I moved them into the `img` subdir within the repo, but it appears that the original location entries have been retained in the DB in addition to the new locations. 

If I look in the database, I can see 2 entries there for each image
- One for the original images that were in the root dir but that have since been moved in github. 
- One for the new image location in the `img` subdir.

If I change the markup so that images are (For Eg) `/img/image1.png` rather than simply `/image1.png` I get the updated version as expected. However the original one still seems to be available even though it's no longer in that location within the repo (I've confirmed that the repo within the docker image is being updated).

**Comment by NGPixel** [comment]
Check the logs during a sync after you moved an asset. It should say "(STORAGE/GIT) Asset marked as deleted: *path/to/the/file*"

Then make sure it's not followed by "Asset not found in the db". This would indicate the old asset was detected as deleted but could not be matched with the existing item in the db.
